conans/test/functional/layout/test_source_folder.py

```diff
@@ -0,0 +1,131 @@
+import pytest
+
+from conans.test.assets.cmake import gen_cmakelists
+from conans.test.assets.genconanfile import GenConanfile
+from conans.test.assets.sources import gen_function_cpp
+from conans.test.utils.scm import create_local_git_repo
+from conans.test.utils.tools import TestClient
+
+
+@pytest.mark.parametrize("no_copy_source", ["False", "True"])
+def test_exports_source_with_src_subfolder(no_copy_source):
```

Comment:
It would be very relevant to learn how this behaves with ``editable`` mode, maybe another test?. I would say there could be issues, as the ``export`` process do change layouts?

Reply:
I would say at `test_editables_layout.py` more than one test is using `self.folders.source` and editable packages. 
The idea still works because the "root" of the editable package is the current project directory, so the `self.folders.source="my_src"` is correct as it is with conan local commands.